watcher/local/watcher_test.go

```diff
@@ -80,6 +84,107 @@ func Test_StartWatching(t *testing.T) {
+			require.NoError(t, statesPub.Publish(txs[1]))
+			require.NoError(t, statesPub.Publish(txs[2]))
+
+			// Trigger events and assert.
+			triggerAdjEventAndExpectNotif(t, trigger, eventsForClient)
+			rs.AssertExpectations(t)
+		})
+		t.Run("happy/newer_than_latest_state_registered", func(t *testing.T) {
+			params, txs := randomTxsForSingleCh(rng, 3)
+			adjSub, trigger := setupAdjudicatorSub(makeRegisteredEvent(txs[2])...)
+
+			rs := &mocks.RegisterSubscriber{}
+			rs.On("Subscribe", mock.Anything, mock.Anything).Return(adjSub, nil)
+			w := newWatcher(t, rs)
+			// Start watching and publish states.
+			statesPub, eventsForClient := startWatchingForLedgerChannel(t, w, makeSignedStateWDummySigs(params, txs[0].State))
+			require.NoError(t, statesPub.Publish(txs[1]))
+
+			// Trigger events.
+			triggerAdjEventAndExpectNotif(t, trigger, eventsForClient)
+
+			rs.AssertExpectations(t)
+		})
+	})
+
+	t.Run("ledger_channel_with_sub_channel", func(t *testing.T) {
+		t.Run("happy/latest_state_registered", func(t *testing.T) {
+			parentParams, parentTxs := randomTxsForSingleCh(rng, 3)
+			childParams, childTxs := randomTxsForSingleCh(rng, 3)
+			parentTxs[2].Allocation.Locked = []channel.SubAlloc{{ID: childTxs[0].ID}} // Add sub-channel to allocation.
+
+			adjSubParent, triggerParent := setupAdjudicatorSub(makeRegisteredEvent(parentTxs[2])...)
+			adjSubChild, triggerChild := setupAdjudicatorSub(makeRegisteredEvent(childTxs[2])...)
+
+			rs := &mocks.RegisterSubscriber{}
+			rs.On("Subscribe", mock.Anything, mock.Anything).Return(adjSubParent, nil).Once()
+			rs.On("Subscribe", mock.Anything, mock.Anything).Return(adjSubChild, nil).Once()
+
+			w := newWatcher(t, rs)
+			// Parent: Start watching and publish states.
+			parentSignedState := makeSignedStateWDummySigs(parentParams, parentTxs[0].State)
+			statesPubParent, eventsForClientParent := startWatchingForLedgerChannel(t, w, parentSignedState)
+			require.NoError(t, statesPubParent.Publish(parentTxs[1]))
+			require.NoError(t, statesPubParent.Publish(parentTxs[2]))
+
+			// Child: Start watching and publish states.
+			childSignedState := makeSignedStateWDummySigs(childParams, childTxs[0].State)
+			statesPubChild, eventsForClientChild := startWatchingForSubChannel(t, w, childSignedState, parentTxs[0].State.ID)
+			require.NoError(t, statesPubChild.Publish(childTxs[1]))
+			require.NoError(t, statesPubChild.Publish(childTxs[2]))
+
+			// Parent, Child: Trigger events.
+			triggerAdjEventAndExpectNotif(t, triggerParent, eventsForClientParent)
+			triggerAdjEventAndExpectNotif(t, triggerChild, eventsForClientChild)
+
+			rs.AssertExpectations(t)
+		})
+		t.Run("happy/newer_than_latest_state_registered", func(t *testing.T) {
+			parentParams, parentTxs := randomTxsForSingleCh(rng, 3)
+			childParams, childTxs := randomTxsForSingleCh(rng, 3)
+			parentTxs[2].Allocation.Locked = []channel.SubAlloc{{ID: childTxs[0].ID}} // Add sub-channel to allocation.
```

Comment:
Why do you only set the alllocs in the last state btw?

Reply:
Because, the last transaction on the ledger channel represents the sub-channel funding.
Since the watching component and mocks used in testing do validate the state, i have not modified the balances.

Reply:
In 67e4a0, added comments to explain that its sub-channel funding transaction.